app/src/main/java/com/example/zingmp3phake/screen/detailplaylist/DetailPlaylistActivity.kt

```diff
@@ -1,17 +1,131 @@
+    BaseActivity<ActivityDetailPlaylistBinding>(ActivityDetailPlaylistBinding::inflate),
+    RecyclerViewRecentAdapter.ItemClickListener,
+    DetailPlaylistContract.View {
 
+    private val adapterRV = RecyclerViewRecentAdapter(this)
+    private val detailPlaylistPresenter = DetailPlaylistPresenter(this)
     override fun initData() {
-        // TODO later
+        val mIntent = intent
+        val bundle = mIntent.getBundleExtra(DATA_KEY)
+        val title = bundle?.getString(BUNDLE_TITLE_STRING_KEY)
+        val list = bundle?.getParcelableArrayList<Song>(BUNDLE_LIST_KEY)
+        binding.textviewNumberOfListSong.text = "${list?.size} $SONG"
+        binding.recyclerView.adapter = adapterRV
+        if (list != null) adapterRV.setData(list)
+        if (title != null) binding.textviewPlaylistName.text = title
+
+        detailPlaylistPresenter.apply {
+            bindService(applicationContext)
+            val filter = IntentFilter(ACTION_MUSIC_BROADCAST)
+            registerReceiver(localReciver, filter)
+        }
     }
 
     override fun initView() {
-        setContentView(R.layout.activity_detail_playlist)
+        binding.apply {
+            buttonBack.setOnClickListener {
+                finish()
+            }
+            buttonNext.setOnClickListener {
+                getPendingIntent(MusicAction.NEXT.name)?.send()
+                binding.buttonPlay.setImageResource(R.drawable.ic_play_24)
+            }
+            buttonPlay.setOnClickListener {
+                getPendingIntent(MusicAction.PLAYORPAUSE.name)?.send()
+            }
+            buttonFavorite.setOnClickListener {
+                getPendingIntent(MusicAction.FAVORITE.name)?.send()
+            }
+            containerStateSong.setOnClickListener {
+                val intent = Intent(applicationContext, DetailSongActivity::class.java)
+                startActivity(intent)
+            }
+        }
+    }
+
+    override fun onItemClick(pos: Int, listSong: MutableList<Song>) {
+        detailPlaylistPresenter.handleStartSong(listSong, pos)
+        binding.buttonPlay.setImageResource(R.drawable.ic_play_24)
+    }
+
+    override fun displayCurrentSong(song: Song) {
+        if (song.songInfo.songid != null) {
+            binding.apply {
+                textviewSongName.text = song.songInfo.songName
+                textviewArtistName.text = song.songInfo.songArtist
+                containerStateSong.visibility = View.VISIBLE
+                if (song.isFavorite) buttonFavorite.setImageResource(R.drawable.ic_favorite_24)
+                else buttonFavorite.setImageResource(R.drawable.ic_unfavorite_24)
+                if (song.isLocal) {
+                    val imgUri = ContentUris.withAppendedId(
+                        Uri.parse(MEDIA_EXTERNAL_AUDIO_URI),
+                        song.songInfo.songImg.toLong()
+                    )
+                    cimg.loadByGlide(applicationContext, imgUri)
```

Comment:
```suggestion
                    cimg.loadByGlide(applicationContext, imgUri)
```
cimg là gì em?
đặt tên tường minh tí đi e